codify-your-developer-vms.md

```diff
@@ -0,0 +1,83 @@
+joining and leaving these projects more or less frequently), developer VMs have become an indispensable tool.
+Not only for the consistency aspect, but also for reducing ramp-up times or being able to easily archive the
+development environment when the project enters the retirement phase.
+
+On regulated projects this is even more relevant, since you need to validate and verify the development toolchain
+and need to ensure that everyone uses exactly that defined toolchain.
+
+## OK, but why should I automate it?
+
+While we started out to craft these VMs manually (and with love) in the early days, we can nowadays easily
+automate the whole procedure with tools like Vagrant, Chef, Ansible and co. Applying automation and
+Infrastructure-as-Code principles to developer VMs not only increases transparency and reproducibility,
+but also allows for an in-VM update mechanism to maintain consistency while the toolchain evolves during
+the lifetime of the project.
+
+Finally, an automated developer VM should yield the following properties:
+
+* updatability - being able to update an existing VM with the push of a button from within the VM
+* testability - running automated tests in the VM to ensure that it works as expected
+* installability - the VM should distributed as a standard VirtualBox / VMware image
+* adaptability - developers should be able to easily adapt and improve the VM as the toolchain evolves
+* reproducibility - changes to the development environment are transparent, traceable and reproducible
+
+## How can I automate it?
+
+The good news is that there are quite a few so-called "Configuration Management" tools out there, which essentially
+allow you to automate the provisioning of software and configuration on top of an existing operating system
+(in our case a fresh installation of the operating system of choice, within a VM image). 
+
+The bad news is that you have to choose one -- and in addition to the configuration management tools that do the
+provisioning, there is (gladly) also a multitude of tools for testing the configuration of such a (virtual) machine.
+
+Assuming you have a preference for either Ruby or Python, these are the tools you will likely end up with:
+
+* Rubyists will like
+  * Chef - as the configuration management tool
+  * Foodcritic - for linting their Chef recipes
+  * InSpec / ServerSpec - for testing the Chef-provisioned systems
+
+* Pythonistas probably prefer
+  * Ansible - as the configuration management tool
+  * ansible-lint - for linting their Ansible roles
+  * TestInfra - for testing the Ansible-provisioned systems
+
+## I’m sold, sounds like pretty cool stuff! What is the fastest way to get there?
+
+It may sound a bit of effort to achieve this if you start from scratch, so that’s why we have a RUA with a
+template / skeleton project along with a tutorial for you. No excuses for doing it manually anymore! ;-)
```

Comment:
```suggestion
template / skeleton project along with a tutorial for you.
```

Reply:
Not everyone will be familiar with the RUA abbreviation

Reply:
@damphyr good point, I removed the reference to RUA

@abeggchr @damphyr I was a bit unsure about this last paragraph in general, considering that the article's content / text is there to stay, while the examples might be broken / non-functional at some point in time. Still they are useful. At least I reduced them to stuff that is hosted on our zuehlke Github account (which is fair to promote I guess).

Still I'm not 100% convinced about referencing these. What's your take on this?